### ISSUE TITLE
Loki: Make label browser accessible from both code and builder modes

### ISSUE THREAD

**Issue by ivanahuckova** [grafana/grafana#57744]
### Summary

Label browser is powerful and praised feature that helps to search trough labels, that is currently accessible and discoverable only trough code mode. To improve accessibility and discoverability, it should be accessible in both - code and builder modes.

### Problem statement

Label browser is very valuable feature that lets you easily navigate through your list of labels and filter and search trough values.  This feature is valuable for users using both - code and builder editor mode, therefore label browser should be accessible from both modes. Moreover, it is currently not discoverable for new users who are using only builder mode.

During user interviews we learned that some users switch between builder-code modes just to use label browser and then go back to builder mode, which is unnecessary. Moreover, this very useful feature is not discoverable for new users who are using only builder mode.

We already have other interactions that work in both modes, so there is not a reason why label browser should not be in builder mode. 

### Scope

Implementation (coding task)

### Solution

In query builder header, we already have a place for features/interactions that are present in both modes (e.g. Kick start your query). We should move label browser button to query builder header and make it accessible in all query modes. 

Moreover, we should evaluate and test if modal wouldn't be a better solution for label browser, as this is what we use in query patterns and would be cohesive. It would also make it much easier to use in dashboard.
